addons/mail/static/tests/messaging_menu/notification_tests.js

```diff
@@ -325,3 +325,23 @@ QUnit.test(
         assert.containsNone($, ".o-mail-MessagingMenu-counter-badge");
     }
 );
+
+QUnit.test(
+    "subtype description should be displayed when body is empty",
+    async (assert) => {
+        const pyEnv = await startServer();
+        const partnerId = pyEnv["res.partner"].create({ name: "Partner1" });
+        const channelId = pyEnv["discuss.channel"].create({ name: "Test" });
+        const subtypeId = pyEnv["mail.message.subtype"].create({ description: "hello" });
+        pyEnv["mail.message"].create({
+            author_id: partnerId,
+            body: "",
+            model: "discuss.channel",
+            res_id: channelId,
+            subtype_id: subtypeId,
+        });
+        await start();
+        await click(".o_menu_systray i[aria-label='Messages']");
+        assert.containsOnce($, ".o-mail-NotificationItem-text:contains(Partner1: hello)");
```

Comment:
Prefer `await contains()`. Synchronous assertion after `await click()` doesn't guarantee the time is right for message fully rendered for what's needed